File: app/go/internal/board/board.go
package board

import "github.com/case-study-game-dev/puyo-puyo-go/internal/puyo"

const (
	// Rows はボードの行数
	Rows = 12
	// Cols はボードの列数
	Cols = 6
)

// Board はぷよぷよの盤面を表す
type Board struct {
	Cells [Rows][Cols]puyo.Color
}

// New は新しいボードを作成する
func New() *Board {
	return &Board{}
}

// Set は指定した位置にぷよを配置する
func (b *Board) Set(row, col int, color puyo.Color) {
	if row >= 0 && row < Rows && col >= 0 && col < Cols {
		b.Cells[row][col] = color
	}
}

// Get は指定した位置のぷよの色を取得する
func (b *Board) Get(row, col int) puyo.Color {
	if row >= 0 && row < Rows && col >= 0 && col < Cols {
		return b.Cells[row][col]
	}
	return puyo.ColorNone
}

File: app/csharp/Assets/Scripts/Core/Board.cs
namespace PuyoPuyo.Core
{
    public class Board
    {
        public const int COLS = 6;
        public const int ROWS = 12;

        private PuyoColor[,] cells;

        public int Cols => COLS;
        public int Rows => ROWS;

        public Board()
        {
            cells = new PuyoColor[ROWS, COLS];
            Clear();
        }

        public PuyoColor Get(int row, int col)
        {
            return cells[row, col];
        }

        public void Set(int row, int col, PuyoColor color)
        {
            cells[row, col] = color;
        }

        public void Clear()
        {
            for (int row = 0; row < ROWS; row++)
            {
                for (int col = 0; col < COLS; col++)
                {
                    cells[row, col] = PuyoColor.None;
                }
            }
        }

        public bool IsInBounds(int row, int col)
        {
            return row >= 0 && row < ROWS && col >= 0 && col < COLS;
        }
    }
}
